src/commands/moderation/createrole.js

```diff
@@ -0,0 +1,34 @@
+const { Command, SwitchbladeEmbed, Constants, Color } = require('../../')
+
+module.exports = class CreateRole extends Command {
+  constructor (client) {
+    super(client, {
+      name: 'createrole',
+      category: 'moderation',
+      requirements: { guildOnly: true, botPermissions: ['MANAGE_ROLES'], permissions: ['MANAGE_ROLES'] },
+      parameters: [
+        { type: 'string', full: false, missingError: 'commands:createrole.noParams', required: true },
```

Comment:
This would not allow people to create roles with spaces on the name. Swap the parameter order and make the string `full: true`.

Reply:
I tried this, but it doesn't work with an optional color parameter. It's either this or that. It should work theoretically, but it seems like a fault lies in underlying parameter parsing code.

Reply:
hm, look at the currency code, it does something like that i think

Reply:
The currency code only takes strings and numbers, while this one takes a color, which is different. I could take 2 strings instead, and parse them myself - that would be a possible solution, but not optimal. What you propose doesn't work, I tried it - what should I do?

Reply:
I just tested it and it worked like a charm

Reply:
What's the default role color?

Reply:
@pedrofracassi - the default role color is white - "#ffffff"

@davipatury - Really? :O I did the exact same, but I get a TypeError. Maybe it has to do with my require() statement? How did you get it to work?

Reply:
```js
const { Command, SwitchbladeEmbed, Constants, Color } = require('../../')

module.exports = class CreateRole extends Command {
  constructor (client) {
    super(client, {
      name: 'createrole',
      category: 'moderation',
      requirements: { guildOnly: true, botPermissions: ['MANAGE_ROLES'], permissions: ['MANAGE_ROLES'] },
      parameters: [
        { type: 'color', required: false },
        { type: 'string', full: true, missingError: 'commands:createrole.noParams', required: true }
      ]
    })
  }

  async run ({ channel, guild, author, t }, color = new Color('#ffffff'), name) {
    const hexcode = color.rgb(true)
    const embed = new SwitchbladeEmbed(author)
    
    try {
      await guild.createRole({ name, color: hexcode })
      embed
        .setTitle(t('commands:createrole.successTitle'))
        .setDescription(t('commands:createrole.successMessage', { name }))
        .setColor(hexcode)
    } catch(err) {
      embed
        .setColor(Constants.ERROR_COLOR)
        .setTitle(t('commands:createrole.errorTitle'))
        .setDescription(`\`${err}\``)
    }

    channel.send(embed)
  }
}
```

Reply:
Thanks. This is the same thing I did, and I now see that the problem was that I had an outdated repo I was testing with, and cloning a fresh repo instead to test the new command worked. Strange. With that, I guess this is resolved.